### ISSUE TITLE
Prelim code review :)

### ISSUE THREAD

**Issue by SethDavenport** [rangle/angular2-ngrx-example#1]
Looks nice - cool theme and idea as well. Overall the code is relatively easy to navigate.

:+1: for ngmodulifying feature areas.

A few suggestions for further discussion:

## Real HTTP requests with a stub service like mocky.io

it would be nice if the services like `videoGamesService`, `platformsService` did actual HTTP requests. That would be clearer than the manually created `Observable` stuff going on. I suggest using a service like http://mocky.io to pre-seed some real HTTP requests that have the same data as what you have mocked out right now.  For example: http://www.mocky.io/v2/5894fbc1290000d4053f42e2 instead of `videoGameResults.ts`

Then `videoGameService.ts` would look like this:

```typescript
import {Injectable} from '@angular/core';
import {Http} from '@angular/http';
import {Observable} from 'rxjs/Observable';
import 'rxjs/add/operator/map';

import {IVideoGame} from '../../interfaces/videoGame/videoGame.interface';

const VIDEO_GAMES_BASE_URL = 'http://www.mocky.io/v2/5894fbc1290000d4053f42e2';

@Injectable()
export class VideoGamesService {
  constructor(private http: Http) {}

  public getAll(): Observable<Array<IVideoGame>> {
    return this.http.get(VIDEO_GAMES_BASE_URL)
      .map(response => response.json());
  }
}
```

This would also give you an opportunity to show 'mergeMap' with an http observable and an effect `action$`, with error handling, which is both awesome and non-obvious.

However even if you don't want to go that far, an easier way to stub an observable is this:

```typescript
import 'rxjs/add/observable/of';

// ...

  public getAll(): Observable<Array<IVideoGame>> {
    return Observable.of(videoGames);
  }
```

Also note my removal of blank lines - I kind of have a (pretend) vendetta against them but I'm not going to make an issue out of it :)

## File name casing

I have an actual vendetta against mixedCaseFileNames. If I had a $100 for each hour spent debugging weird CI and deployment issues caused by subtle differences in how OSX, Linux, and Windows treat file name case I would have like.... $1500 :).  Anyway. I've been pushing for `kebab-case-all-the-file-names.component.ts` for a while now in both the Angular and React sides of the business. For fonts, images, and css too.

## Effects location

Currently they're under store, with the reducers. Not sure if they should be in the feature folders instead.  If this were React, a large chunk of the logic would be in sagas (the analogous construct). It would be nice if this logic were lazy loadable along route lines.

Probably worth more discussion though.

## Folder structure

I mostly like it but a couple of things confused me:

`app/components/app`: I would tend to put app in it's own top-level folder.  Something like:

* src
* src/app/app.component.html etc.
   * app.module and app.routes in here too.
* src/store
* src/shared
* src/video-games

Alternately if you want to keep 'app' as the top level folder name instead of `src`, maybe something like:

* app
* app/root/root.component.html etc.
   * root.module and root.routes in here too.
* app/store
* app/shared
* app/video-games

## Interfaces folder.

I'm not sure how I feel about this - I like to co-locate all the things along feature lines.  Feature specific interfaces in the feature folders, store-general interfaces in `store`. What do you think?

**Comment by egervari** [comment]
Thanks for taking the time to do this. Honestly this is a great code review.

So far...

1. Added root module
2. Added mock endpoints. This is a great idea. I had no idea we could do this!!! Like holy cow, that's amazing! :) I actually went further here and created ApiService to refactor out the duplication in both the services
3. I'm going to keep the blank lines - my vision is terrible and so the lines help me predict what's there without actually having to squint and read it. This is why my code is so clean in general... because I'm blind in one eye and I'm 20/400 in the other, so I don't really have a choice but to use every trick under the sun to help me out. Just some FYI for why I do these things. It's not because I am opinionated in a general sense - I just have to do it this way.
4. Changed file name casing
5. The interfaces stuff I completely agree with. I first put them in Store, because I wasn't entirely where they would be used. It was on my mind though - totally in agreement. Fixed.
6. Effects/Reducers/Store locations - Fixed

All done
